### ISSUE TITLE
Investigate Potential Bug in MCMC Tree Moves

### ISSUE THREAD

**Issue by gordonkoehn** [cbg-ethz/PYggdrasil#48]
During the visualization issue #42 some trees in an outdated, hence now irreproducible dataset, outputted by the MCMC sampler were not convertible to the `TreeNode` format. The reason was, that they seemed to lack a root. - Or the last node in the adjacency matrix at least was not the root. The exact issue is not clear. The function `tree_inference_tree._get_root_label` got an empty array in their line 
`# find row which has all ones in ancestor_matrix
    root_idx = jnp.where(jnp.all(ancestor_matrix == 1, axis=1))[0]`
    
- [x] Consider adding debug logging to the tree moved to discover what went wrong the next time this happens.
- [x] Add tests for the individual tree moves and check if it is a valid tree
- [x] identify the issue
- [x] fix the issue
- [x] rerun mcmc mark00
- [x] check if all trees can be viszualized
- [x] check speed of mcmc

Conjectures:
- root is assumed to be the last in the tree, and was swapped to be not the last
- confusion if to include the root in moves or not

**Comment by gordonkoehn** [comment]
Write a test now to check the nested subtree case as Figure 17 in the supplement of the SCITE paper. As expected it is failed. 

There is just some logic missing here. Next: 
- [ ] consider what the proposal has to check for - if they are of the same lineage? or just weather to swap the nodes, if one is a descendant of the other? 
-  [ ] add logic to the `swap_subtree_move`

**Comment by gordonkoehn** [comment]
Implemented first not-working version of the complete logic of `swap_subtrees` 

Next: debug with a print statement -already worked in.

**Comment by gordonkoehn** [comment]
New runtime per sample: 269ms at $n_{cells} = 2000$ and $n_{mutations}=50$
